explainers/subresource-loading.md

```diff
@@ -0,0 +1,101 @@
+# Explainer: Subresource loading with Web Bundles
+
+Last updated: March 2020
+
+We propose a new approach to load a large number of resources efficiently using
+a format that allows multiple resources to be bundled, e.g.
+[Web Bundles](https://web.dev/web-bundles/).
+
+## Backgrounds
+
+- Loading many unbundled resources is still slower in 2020. We concluded that
+  [bundling was necessary in 2018](https://v8.dev/features/modules#bundle), and
+  our latest local measurement still suggests that.
+
+- Caching stories with JS bundlers (e.g. webpack) is tough. They are pretty good
+  tools but configuring them to work in an optimal way is tough, and sometimes
+  also conflicts with new requirements like dynamic bundling (e.g. small edit
+  with tree shaking could invalidate everything).
+
+- With JS bundlers, execution needs to wait for the full bytes to come. Ideally
+  loading multiple subresources should be able to utilize full streaming and
+  parallelization, while it's not possible if all resources are bundled as one
+  javascript. (For JS modules execution still needs to be waited for the entire
+  tree due to the current
+  [deterministic execution model](https://docs.google.com/document/d/1MJK0zigKbH4WFCKcHsWwFAzpU_DZppEAOpYJlIW7M7E/edit#heading=h.5652gd5ks5id))
+
+- Related issues: [#411](https://github.com/WICG/webpackage/issues/411),
+  [#526](https://github.com/WICG/webpackage/issues/526)
+
+## Proposal
+
+We propose a new `link rel=”webbundle”` notion [**1**] that lets the browser
+fetch bundled resources and attach them to the page. Such as:
+
+```html
+<link
+  rel="webbundle"
+  href="https://example.com/dir/subresources.wbn"
+  scope="https://example.com/scope/"
+/>
+```
+
+If the main document has such a `link` element, a browser must do:
+
+- A browser must fetch the specified Web Bundle, `subresources.wbn`.
+
+- All the subresources whose URLs are under the scope
+  `https://example.com/scope/` must [**2**] be loaded from the specified Web
```

Comment:
The primary requirement to avoid fetching the same bytes twice is that "If a subresource under that scope is needed later in the document, that later fetch should block until at least the index of the bundle has downloaded to see if it's there."

It seems secondary to then say that if the subresource within that scope isn't in the bundle, its fetch should fail or otherwise notify the developer: that just prevents delays in starting the subresource fetch.

Reply:
Thanks! Let me add these almost "as-is".  I think that would help the reader of this explainer to understand the possible behaviors we are thinking in more details.